pkgs/by-name/cl/clash-verge-rev/service.nix

```diff
@@ -1,47 +1,30 @@
 {
-  version,
   rustPlatform,
-  src-service,
-  pkg-config,
-  openssl,
-  pname,
-  service-cargo-hash,
+  fetchFromGitHub,
   meta,
 }:
-rustPlatform.buildRustPackage {
-  pname = "${pname}-service";
-  inherit version meta;
 
-  src = src-service;
-  sourceRoot = "${src-service.name}";
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "clash-verge-service-ipc";
+  version = "2.0.21";
 
-  patches = [
-    # I want to keep these patches because it's not harmful.
-
-    # Patch: Restrict bin_path in spawn_process to be under the clash-verge-service directory.
-    # This prevents arbitrary code execution by ensuring only trusted binaries from the Nix store are allowed to run.
-    ./0001-core-validate-bin_path-to-prevent-RCE-in-start_clash.patch
+  src = fetchFromGitHub {
+    owner = "clash-verge-rev";
+    repo = "clash-verge-service-ipc";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-9c9fM1l31NbY//Ri50Ql60BWWgISjMWj72ABixRaXvM=";
+  };
 
-    # Patch: Add validation to prevent overwriting existing files.
-    # This mitigates arbitrary file overwrite risks by ensuring a file does not already exist before writing.
-    ./0002-core-prevent-overwriting-existing-file-by-validating.patch
+  postPatch = ''
+    substituteInPlace src/lib.rs \
+      --replace-fail "/tmp/verge/clash-verge-service.sock" "/run/clash-verge-rev/service.sock"
+  '';
 
-    # Patch: move IPC directory from /tmp to /run/clash-verge-rev/service.lock
-    # This allows we enable ProtectSystem="strict" and PrivateTmp
-    ./0003-IPC-move-path-to-run-clash-verge-rev-service.sock.patch
+  cargoHash = "sha256-UbNN3uFu5anQV+3KMFPNnGrCDQTGb4uC9K83YghfQgY=";
+  buildFeatures = [
+    "standalone"
   ];
-
-  nativeBuildInputs = [
-    pkg-config
-  ];
-
-  buildInputs = [
-    openssl
-  ];
-
-  env = {
-    OPENSSL_NO_VENDOR = 1;
-  };
-
-  cargoHash = service-cargo-hash;
-}
+  # tests are broken
```

Comment:
It would be helpful to have more explanation on how broken.

Reply:
After further investigation, I finally made the tests work. The tests calls another binary as a client, so I have to build it in advance.